test/unit/update-test.js

```diff
@@ -0,0 +1,61 @@
+var simple = require('simple-mock')
+var test = require('tape')
+
+var update = require('../../lib/update')
+
+test('update without change', function (t) {
+  t.plan(1)
+
+  update()
+    .then(t.fail.bind(t, 'must reject'))
+    .catch(t.pass.bind(t, 'rejects with error'))
+})
+
+test('update with change', function (t) {
+  t.plan(3)
+
+  simple.mock(update.internals, 'request').resolveWith({
+    statusCode: 204,
+    body: null
+  })
+  simple.mock(update.internals, 'serialise').returnWith('jsonData')
+  simple.mock(update.internals, 'saveAccount').callFn(function () {})
+
+  update({
+    cacheKey: 'cacheKey123',
+    url: 'http://example.com',
+    account: {
+      username: 'bakingpies',
+      session: {
+        id: 'abc1234'
+      }
+    }
+  }, {
```

Comment:
can you please store the first argument in `var state = { ... }` and then do `update(state, { username: 'treetrunks' })`, that will make it more clear